docs/source/guide/custom_metric.md

```diff
@@ -105,4 +107,137 @@ After adding your code to Label Studio Enterprise, the following could happen:
+            "Effect": "Allow",
+            "Action": "logs:CreateLogGroup",
+            "Resource": "arn:aws:logs:*:YOUR_AWS_ACCOUNT:*"
+        },
+        {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": [
+                "logs:CreateLogStream",
+                "logs:PutLogEvents"
+            ],
+            "Resource": [
+                "arn:aws:logs:*:YOUR_AWS_ACCOUNT:log-group:/aws/lambda/custom-metric-*"
+            ]
+        }
+    ]
+}
+```
+
+### Define an IAM policy
+
+After creating an IAM role to manage logs for the custom agreement metric, define an IAM policy to allow Label Studio Enterprise to interact with AWS Lambda. If you haven't created an IAM policy in AWS before, review the AWS Identity and Access Management documentation on [Creating IAM policies](https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_create.html). 
+
+Create an IAM policy called `LSE_AllowInteractLambda` and replace `YOUR_AWS_ACCOUNT` with your AWS account ID:
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": "iam:PassRole",
+            "Resource": "arn:aws:iam::YOUR_AWS_ACCOUNT:role/LSE_CustomMetricsExecuteRole"
+        },
+        {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": [
+                "lambda:CreateFunction",
+                "lambda:UpdateFunctionCode",
+                "lambda:InvokeFunction",
+                "lambda:GetFunction",
+                "lambda:DeleteFunction"
+            ],
+            "Resource": [
+      "arn:aws:lambda:*:YOUR_AWS_ACCOUNT:function:custom-metric-*"
+            ]
+        },
+        {
+            "Sid": "VisualEditor2",
+            "Effect": "Allow",
+            "Action": "lambda:ListFunctions",
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+How you attach the policy to the proper `label-studio` role depends on how you deployed Label Studio Enterprise in your self-managed cloud infrastructure. 
+- [Deployed with Docker Compose running in EC2](#Deployed-with-Docker-Compose-running-in-EC2)
+- [Deployed in EKS with an OIDC provider](#Deployed-in-EKS-with-an-OIDC-provider)
+- [Deployed in EKS without an OIDC provider](#Deployed-in-EKS-without-an-OIDC-provider)
+
+### Deployed with Docker Compose running in EC2
+
+If you deployed Label Studio Enterprise using Docker Compose in an AWS EC2 instance, do the following to finish setting up permissions for the custom agreement metric functionality:
+1. Create an IAM user with an Access key with Programmatic access. Save the user credentials somewhere secure.
+2. Attach the policy created in the previous step, `LSE_AllowInteractLambda` to the IAM user that you create.
+3. In the `docker-compose.yaml` file that you use to deploy Label Studio Enterprise, add the following environment variables in the `app` and `rqworkers` sections:
```

Comment:
should we provide a more detailed example, or is this straightforward enough do you think?

Reply:
ref to the separate comment in this PR